### ISSUE TITLE
Is the answer start considered for training ?

### ISSUE THREAD

**Issue by gyan7611** [andrejonasson/question_answering#5]
Hi ,
You have considered the final loss as the average of the answer start and  answer end loss. Does this preserve the ordering ? Because when I test it on some data , the start index of the answer is greater than the index of the end, eg: (1768 , 3) and the index exceeds the length of the sentence . Eg: The data has only 8 words while the answer is (1768,3). 

It would really be helpful if you could clarify the doubt.

**Comment by andrejonasson** [comment]
https://arxiv.org/pdf/1711.00106.pdf

Section 2.2, first equation shows you the kind of loss they use for the cross entropy part. There is no ordering in the loss, everything is summed into one number and then it backpropagates into all the iterations, and since the start logit is part of the sum it will backpropagate through the answer start logit as well.

It concerns me that you are getting an index greater than the document length. Could you give me some more information? Like when you updated the code last. Some updates included masking of the logits with `-1e30` at positions greater than `document_length`. When I run the model on the dev set I only get starts and ends that are less than the document length.

`start > end` is definitely possible there's nothing in the code that forces `start <= end`.

Edit: I added some features that force "end >= start" (optional) and end <= start + max_answer_length (optional). Be sure to turn these off if you test if part of your answer is greater than document_length. If you still get that some answer is greater, please report it here.

**Comment by gyan7611** [comment]
Thanks ,
The `start < end` looks fixed . However , I'm still getting the indices > `document_length` in some cases . I'm currently exploring as to how it's happening (thinking it's a bug that I created for my implementation.) I will drop my findings as I get them .

Thanks again for being proactive about this

**Comment by Abinaya132** [comment]
Just leaving a comment here although not sure if it is related to the issue. 
Hi @andrejonasson , I am training on the same data (SQUAD) for 40 epochs and the minimum loss was 10 at step 20. But after that , the loss is lurking in the range of 10^29 . When you trained on your data , what was the minimum loss and can u share some sort of performance metric and the batch size & the # of  epochs which you found had the best performance?

Would be useful if u could clarify this.

**Comment by andrejonasson** [comment]
@Abinaya132 My first guess would be that your learning rate is too large or that your batch size is too small and that the training diverges because of it. Did you try running the baseline model with the settings in the README file?

You should be able to get ~3 average cross entropy without too much effort.

If this doesn't work, copy paste the flags JSON file content from the checkpoints folder here and I'll have a look at your hyperparams.

**Comment by Abinaya132** [comment]
Ok thanks . I was running with a batch size of 512. And the learning rate was the same as yours.. Let me try playing with the batch size and run the baseline model .
